integ/src/main.rs

```diff
@@ -64,27 +65,55 @@ struct IntegrationTestArgs {
     instance_ami_id: String,
 }
 
-async fn generate_kube_config_path(args_kube_config_path: &str) -> Result<String> {
-    // default kube config path is /temp/kubeconfig.yaml
-    let kube_config_path = match args_kube_config_path {
-        DEFAULT_KUBECONFIG_FILE_NAME => format!(
-            "{}/{}",
-            temp_dir().to_str().context(error::FindTmpDir)?,
-            DEFAULT_KUBECONFIG_FILE_NAME
-        ),
-        res => res.to_string(),
-    };
+async fn generate_kubeconfig(arguments: &Arguments) -> Result<String> {
+    // default kube config path is /temp/{CLUSTER_NAME}-{REGION}/kubeconfig.yaml
+    let kube_config_path = generate_kubeconfig_file_path(&arguments).await?;
+
+    // decode and write kubeconfig
+    info!("decoding and writing kubeconfig ...");
+
+    write_kubeconfig(
+        &arguments.cluster_name,
+        &arguments.region,
+        &kube_config_path,
+    )
+    .context(error::WriteKubeconfig)?;
+    info!(
+        "kubeconfig has been written and store at {:?}",
+        &kube_config_path
+    );
 
     Ok(kube_config_path)
 }
 
+async fn generate_kubeconfig_file_path(arguments: &Arguments) -> Result<String> {
+    let unique_kube_config_temp_dir = get_kube_config_temp_dir(&arguments)?;
+
+    fs::create_dir_all(&unique_kube_config_temp_dir).context(error::CreateDir)?;
+
+    let kube_config_path = format!(
+        "{}/{}",
+        &unique_kube_config_temp_dir, DEFAULT_KUBECONFIG_FILE_NAME
+    );
+
+    Ok(kube_config_path)
+}
+
+fn get_kube_config_temp_dir(arguments: &Arguments) -> Result<String> {
```

Comment:
nit: Could make this a bit more specific to clarify that it acts on a path rather than a file.

```suggestion
fn get_kube_config_temp_dir_path(arguments: &Arguments) -> Result<String> {
```